### PR TITLE
[lldb] Handle diagnostics better around expression evaulation retries

### PR DESCRIPTION
**Pull request by augusto2112**
The error message produced when expression evaluation parsing failed with automatic retries was not great, and would confuse many users. Change the logic around what diagnostics should be displayed to the user if expression evaluation parsing retry also fails to show the original error message.

rdar://118057664

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

**Review by adrian-prantl — COMMENTED** [review]
I'm generally fine with the approach, I would just like you to double-check that SwiftASTContext::GetScopedDiagnosticConsumer() isn't better suited for the job.

**Comment by adrian-prantl** [comment]
An maybe the answer is no

**Comment by adrian-prantl** [comment]
Do you have a testcase?

**Comment by augusto2112** [comment]
> I'm generally fine with the approach, I would just like you to double-check that SwiftASTContext::GetScopedDiagnosticConsumer() isn't better suited for the job.

As I understand it, SwiftASTContext::GetScopedDiagnosticConsumer is used to capture diagnostics generated by the swift compiler, in this case we're dealing with the more general lldb diagnostics.

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@adrian-prantl had to change my approach a little, if we decide this is the way to go I'll upstream the patch that adds `DiagnosticManager::Consume` before merging this one.

**Comment by augusto2112** [comment]
@swift-ci test windows

File: lldb/test/API/lang/swift/fixits/TestSwiftFixIts.py [hunk 1]
```diff
@@ -65,21 +65,6 @@ def do_test(self):
         options = lldb.SBExpressionOptions()
         options.SetAutoApplyFixIts(False)
 
-        # First make sure the expressions fail with no fixits:
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/fixits/TestSwiftFixIts.py]: Why are these removed?

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/test/API/lang/swift/private_generic_type/TestSwiftPrivateGenericType.py [hunk 2]
```diff
@@ -151,3 +151,8 @@ def test_private_generic_type(self):
                     substrs=["Could not evaluate the expression without binding generic types."], 
                     error=True)
 
+        # Check that if both binding and not binding the generic type parameters fail, we report 
+        # the "bind generic params" erroe message, as that's the default case that runs first.
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/private_generic_type/TestSwiftPrivateGenericType.py]:
```suggestion
        # the "bind generic params" error message, as that's the default case that runs first.
```

**Review by adrian-prantl — APPROVED** [review]

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test Windows

**Comment by augusto2112** [comment]
@swift-ci test macOS

**Comment by augusto2112** [comment]
@swift-ci test Windows

**Comment by augusto2112** [comment]
@swift-ci test macOS

**Comment by augusto2112** [comment]
@swift-ci test Windows

**Comment by adrian-prantl** [comment]
@swift-ci test macos

**Comment by adrian-prantl** [comment]
@swift-ci test macos
